components/cam/src/physics/cam/shoc.F90

```diff
@@ -9,11 +7,59 @@ module shoc
 ! email: bogenschutz1@llnl.gov
 !--------------------------------------------------------------
 
+#define bfb_square(val) ((val)*(val))
+#define bfb_cube(val) ((val)*(val)*(val))
+
+#ifdef SCREAM_CONFIG_IS_CMAKE
```

Comment:
I'm seeing a lot of repeated code. Since micro_p3 and shoc live side-by-side, it should be easy to have common headers etc.